ciao-controller/controller_test.go

```diff
@@ -732,22 +731,21 @@ func testStartWorkloadLaunchCNCI(t *testing.T, num int) (*testutil.SsntpTestClie
 	newTenant := uuid.Generate().String() // random ~= new tenant and thus triggers start of a CNCI
 
 	// trigger the START command flow, and await results
-	// NOTE: "instances" is shared with the go subroutine and we must
-	// insure consistency between parent and child processes
-	var instances []*types.Instance
+	instanceCh := make(chan []*types.Instance, 1)
 
-	var wg sync.WaitGroup
-	wg.Add(1)
 	go func() {
-		instances, err = context.startWorkload(wls[0].ID, newTenant, 1, false, "")
+		var instances []*types.Instance
+
+		instances, err := context.startWorkload(wls[0].ID, newTenant, 1, false, "")
 		if err != nil {
 			t.Fatal(err)
 		}
 
 		if len(instances) != 1 {
 			t.Fatalf("Wrong number of instances, expected 1, got %d", len(instances))
 		}
-		wg.Done()
+
+		instanceCh <- instances
```

Comment:
Does the race detector identify instances as the problem?Could something else be racy instead, e.g., context.ds?  The TenantCNCI.Subnets slice maybe?    The original code seems to protect instances as you already mention, although this code is a bit nicer I think.